src/NuGet.Services.AzureSearch/AuxiliaryFiles/DownloadDataClient.cs

```diff
@@ -41,36 +41,45 @@ public class DownloadDataClient : IDownloadDataClient
 
         private ICloudBlobContainer Container => _lazyContainer.Value;
 
-        public async Task<ResultAndAccessCondition<DownloadData>> ReadLatestIndexedAsync()
+        public async Task<AuxiliaryFileResult<DownloadData>> ReadLatestIndexedAsync(IAccessCondition accessCondition)
         {
             var stopwatch = Stopwatch.StartNew();
             var blobName = GetLatestIndexedBlobName();
             var blobReference = Container.GetBlobReference(blobName);
 
             _logger.LogInformation("Reading the latest indexed downloads from {BlobName}.", blobName);
 
+            bool notModified;
             var downloads = new DownloadData();
-            IAccessCondition accessCondition;
+            AuxiliaryFileMetadata metadata;
             try
             {
-                using (var stream = await blobReference.OpenReadAsync(AccessCondition.GenerateEmptyCondition()))
+                using (var stream = await blobReference.OpenReadAsync(accessCondition))
                 {
-                    accessCondition = AccessConditionWrapper.GenerateIfMatchCondition(blobReference.ETag);
                     ReadStream(stream, downloads.SetDownloadCount);
+                    notModified = false;
```

Comment:
>notModified = false; [](start = 20, length = 20)

This `notModified = false;` hurts my brain a little (or maybe it's lack of sleep?), what do you think of renaming everything to `modified` to avoid the double negation?

Reply:
Maybe add comments on `AuxiliaryFileResult`'s properties if you do this change

---
In reply to: [314494827](https://github.com/NuGet/NuGet.Services.Metadata/pull/632#discussion_r314494827) [](ancestors = 314494827)